firmware/application/src/settings.h

```diff
@@ -0,0 +1,21 @@
+#ifndef SETTINGS_H
+#define SETTINGS_H
+
+#include <stdint.h>
+
+#define SETTINGS_ANIMATION_FULL 0
+#define SETTINGS_ANIMATION_MINIMAL 1
+#define SETTINGS_ANIMATION_NONE 2
+
+/*
+ * bits [0-1]: animation config
+ * bits [2-31]: reserved
+ */
+typedef uint32_t settings_data_t;
```

Comment:
Probably a bitfield struct would be cleaner imho.
There's should also be a `version` field (8 bit should be enough? 16 to be sure?) so that settings can be cleanly upgraded.

Reply:
Wow, nice, I didn’t know of bitfield struct. After the weekend I will
rework it this way :)
And version field - I can add it too, although I am not too sure if there’s
really need for it, as long as we ensure that default value will be 0x00
after firmware update it will take on this value.

Reply:
> And version field - I can add it too, although I am not too sure if there’s really need for it, as long as we ensure that default value will be 0x00 after firmware update it will take on this value.

Version field is useful if a setting changes value range, meaning or gets removed

Reply:
Added version field and made it bitfield struct, thus resolving your comment :)